src/background/wallet/wallet-utils.ts

```diff
@@ -0,0 +1,73 @@
+import { Account, deriveAccount, generateWallet, WalletConfig } from '@stacks/wallet-sdk';
+import { Wallet as SDKWallet } from '@stacks/wallet-sdk';
+import { mnemonicToSeed } from 'bip39';
+import { fromSeed } from 'bip32';
+import { BIP32Interface } from 'bitcoinjs-lib';
+
+function getSavedWalletConfig() {
+  const walletConfig = localStorage.getItem('walletConfig');
+  if (typeof walletConfig !== 'string') return;
+  return JSON.parse(walletConfig) as WalletConfig;
+}
+
+function accountsFromWalletConfig(
+  walletConfig: WalletConfig,
+  rootNode: BIP32Interface,
+  salt: string
+) {
+  return walletConfig.accounts.map((account, index) => {
+    const existingAccount = deriveAccount({
+      rootNode,
+      index,
+      salt,
+    });
+    return {
+      ...existingAccount,
+      username: account.username,
+    };
+  });
+}
+
+async function rootNodeFromSecretKey(secretKey: string) {
+  const rootPrivateKey = await mnemonicToSeed(secretKey);
+  return fromSeed(rootPrivateKey);
+}
+
+async function getSavedWalletAccounts({
+  secretKey,
+  walletConfig,
+  salt,
+}: {
+  secretKey: string;
+  walletConfig: WalletConfig;
+  salt: string;
+}): Promise<Account[]> {
+  // Restore from existing config
+  const rootNode = await rootNodeFromSecretKey(secretKey);
+  if (!walletConfig) return [] as Account[];
+  return accountsFromWalletConfig(walletConfig, rootNode, salt);
+}
+
+export async function getWallet(
+  secretKey: string,
+  salt: string,
+  password: string
+): Promise<SDKWallet | undefined> {
```

Comment:
For consistency, could this be an object too, as above? Somewhat stylistic, but imo for readability, it's clearer to have a separate interface for the arguments, rather than defining it inline.